### ISSUE TITLE
Option + Left/Right/Delete don't work in the terminal window

### ISSUE THREAD

**Issue by nchammas** [Microsoft/vscode#15933]
The terminal window is great. However, the navigation shortcuts to navigate one word forward or back don't appear to work. Instead of jumping forward and back one word, they print the following characters:

* Option + Left: `;5D`
* Option + Right: `;5C`

Option + Delete also doesn't delete a word at a time, and instead just behaves as a regular Delete, deleting one character at a time.

These shortcuts work normally in regular text boxes (e.g. here on GitHub) or in macOS Terminal when "Use Option as Meta key" is enabled.

Environment:
- VSCode Version: 1.7.2
- OS Version: macOS 10.12.1

**Comment by Tyriar** [comment]
Duplicate https://github.com/sourcelair/xterm.js/issues/225

**Comment by nchammas** [comment]
@Tyriar - Thanks for the pointer to the upstream issue.

Technically, this is not a duplicate since this is an issue affecting VS Code, even though the root issue is upstream.

After the upstream issue is fixed, how will users know when that fix is included in VS Code? Would this issue not be the correct place to communicate that?

Don't mean to nitpick on issue management; I'm just curious as to how things are supposed to work.

**Comment by Tyriar** [comment]
@nchammas good question :smile:

I've been using a sort of hybrid approach for managing issues around the terminal to date, sometimes having the issue in the vscode repo if it's important, and not if it's not so important. It turns out this is a huge pain to manage all these different issue so I've decided to move to a system where if the issue will be fixed upstream and all that needs to be done is upgrade the dependency then I close it in vscode. If there will be additional integration work necessary to get the fix I keep it open in vscode.

The reason I'm moving to a system like this is because I spend a big chunk of my time contributing to xterm.js, so much so that it's linked as a dependency to vscode by pinning a commit, not using the stable version. I do this because of my familiarity with it, to dogfood recent changes and because I have commit access on xterm.js as well so I can freely manage the issues over there.

Due to this change I've started going through the commits for xterm.js for each release and pulling in notable fixes/features that came through upstream in the release notes. I also think it's beneficial to more centralize the discussion upstream and to also give the other owners of that xterm.js more insight into how many people are experiencing that particular issue (by count of closed issue links).
